web/src/app-provided.tsx

```diff
@@ -0,0 +1,91 @@
+// does most of what index.js does but as a component
+// export component
+// pass in websockets as props
+
+import React, { useState } from 'react'
+import { useEffect, useRef } from "react"
+import { applyMiddleware, createStore, compose } from "redux"
+import { Provider } from 'react-redux'
+import App from './routes/App.connector'
+import { layoutWatcher } from './redux/ephemeral/layout/middleware'
+import { realtimeInfoWatcher } from './redux/persistent/projects/realtime-info-signal/middleware'
+import acorn from './redux/reducer'
+import signalsHandlers from './signalsHandlers'
+import { MAIN_APP_ID, PROFILES_ROLE_ID } from './holochainConfig'
+import { getAppWs, setAgentPubKey } from './hcWebsockets'
+import { cellIdToString } from './utils'
+import { setProfilesCellId, setProjectsCellIds } from './redux/persistent/cells/actions'
+import ProfilesZomeApi from './api/profilesApi'
+import { fetchAgents } from './redux/persistent/profiles/agents/actions'
+import { whoami } from './redux/persistent/profiles/who-am-i/actions'
+import { fetchAgentAddress } from './redux/persistent/profiles/agent-address/actions'
+import { getProjectCellIdStrings } from './projectAppIds'
+
+
+function AppProvided({
+    appWs,
+    adminWs,
+}) {
+    const [storeLoaded, setStoreLoaded] = useState(false)
+    const middleware = [layoutWatcher, realtimeInfoWatcher]
+
+    // This enables the redux-devtools browser extension
+    // which gives really awesome debugging for apps that use redux
+    // @ts-ignore
+    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
+
+    // acorn is the top-level reducer. the second argument is custom Holochain middleware
+    let initialStore = createStore(
+        acorn,
+  /* preloadedState, */ composeEnhancers(applyMiddleware(...middleware))
+    )
+    const store = useRef(initialStore)
+
+    // initialize the appWs with the signals handler
+    const signalCallback = signalsHandlers(store.current)
+
+    useEffect(() => {
+        const prepareStore = async () => {
+            getAppWs(signalCallback).then(async (client) => {
```

Comment:
we will want to replace this line

Reply:
Is it possible to hook up a signal callback to an already instantiated `AppWebsocket`?

Reply:
Lol... no

Check out the create date on this
https://github.com/holochain/holochain-client-js/issues/37

However, it is not out of the question to open a second AppWebsocket in order to subscribe to signals, by introspecting (accessing nested properties) on the AppWebsocket we are passed so as to connect another new one to the same websocket port